docs/source/array-best-practices.rst

```diff
@@ -0,0 +1,91 @@
+Best Practices
+==============
+
+It is easy to get started with Dask arrays, but using them *well* does require
+some experience.  This page contains suggestions for best practices, and
+includes solutions to common problems.
+
+Use NumPy
+---------
+
+If your data fits comfortably in RAM and you are not performance bound, then
+using NumPy might be the right choice.  Dask adds another layer of complexity
+which may get in the way.
+
+
+Select a good chunk size
+------------------------
+
+A common performance problem among Dask Array users is that they have chosen a
+chunk size that is either too small (leading to lots of overhead) or poorly
+aligned with their data (leading to inefficient reading).
+
+While optimal sizes and shapes are highly problem specific, it is rare to see
+chunk sizes below 100 MB in size.  If you are dealing with float64 data then
+this is around ``(4000, 4000)`` in size.
+
+
+Orient your chunks
+------------------
+
+When reading data you should align your chunks with your storage format.
+Most array storage formats store data in chunks themselves.  If your Dask array
+chunks aren't multiples of these chunk shapes then you will have to read the
+same data repeatedly, which can be very expensive.  Note though that often
+storage formats choose chunk sizes that are much smaller than is ideal for
```

Comment:
I've mentioned that many storage technologies choose something in the 1MB range, but I don't mention NetCDF in particular in case it confuses the non-earth-science folks.

Reply:
Yep, also agree this looks sufficient as is.